include/meltpooldg/melt_pool/melt_pool_operation.hpp

```diff
@@ -167,16 +273,60 @@ namespace MeltPoolDG
             {
               cell->get_dof_indices(local_dof_indices);
               for (unsigned int i = 0; i < dofs_per_cell; ++i)
-                temperature[local_dof_indices[i]] =
-                  analytical_temperature_field(support_points[local_dof_indices[i]],
-                                               level_set_as_heaviside[local_dof_indices[i]]);
+                {
+                  temperature[local_dof_indices[i]] =
+                    analytical_temperature_field(support_points[local_dof_indices[i]],
+                                                 level_set_as_heaviside[local_dof_indices[i]]);
+                  solid[local_dof_indices[i]] =
+                    is_solid_region(support_points[local_dof_indices[i]]) ? 1.0 : 0.0;
```

Comment:
```suggestion
                  solid[local_dof_indices[i]] = is_solid_region(support_points[local_dof_indices[i]]);
```
... if you don't want implicit conversion, you could also introduce a static cast (what we do in deal.II).